src/components/marker.js

```diff
@@ -48,11 +53,131 @@ const defaultProps = Object.assign({}, BaseControl.defaultProps, {
  * recalculate the marker's position when the parent re-renders.
  */
 export default class Marker extends BaseControl {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dragPos: null,
+      dragOffset: null
+    };
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount();
+    this._removeDragEvents();
+  }
+
+  _setupDragEvents() {
+    const {eventManager} = this.context;
+    if (!eventManager) {
+      return;
+    }
+
+    // panstart is already attached by parent class BaseControl,
+    // here we just add listeners for subsequent drag events
+    this._dragEvents = {
+      panmove: this._onDrag.bind(this),
+      panend: this._onDragEnd.bind(this),
+      pancancel: this._onDragCancel.bind(this)
+    };
+    eventManager.on(this._dragEvents);
+  }
```

Comment:
These events need to be bound to the map itself (instead of the marker), otherwise you get behavior like this when the mouse moves outside the pin:

![draggable-panmove](https://user-images.githubusercontent.com/875591/43994183-7e8403e8-9d4d-11e8-86f5-4a8864dff9c2.gif)

And it makes sense to attach these events _after_ panstart, otherwise with 1000 markers you'd have 1000 onDrag event handlers being invoked after each panmove.